### PR TITLE
Add Bytes support to csnex

### PR DESCRIPTION
**Pull request by gitlarryf**

### CODE REVIEW

File: exec/csnex/Extensions.cs [hunk 1]
```diff
@@ -0,0 +1,36 @@
+﻿namespace csnex
+{
+    internal static partial class CodeExtensions
+    {
+        public static int Compare(this byte[] self, byte[] rhs)
```
Comment (ghewgill) [inline exec/csnex/Extensions.cs:5]: Why is this function written this way, as a method? Shouldn't it just be an ordinary function that takes two `byte[]` parameters and returns an `int` result? Not everything needs to be a method.

File: exec/csnex/Extensions.cs [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+                return 1;
+            } else if (self == null && rhs != null) {
+                if (rhs.Length == 0) {
+                    return 0;
+                }
+                return -1;
+            }
+
+            int res = 0;
+            for (int i = 0; i < self.Length && i < rhs.Length; i++) {
```
Comment (ghewgill) [inline exec/csnex/Extensions.cs:22]: I don't think this works correctly as written. Compare two strings where one is a prefix of the other, like `car` and `carpet`. What should the result of this function be?

**Review by ghewgill — CHANGES_REQUESTED** [review]

File: exec/csnex/Extensions.cs (hunk 2 above)
Comment (gitlarryf) [reply exec/csnex/Extensions.cs]: Rewrote to support better logic.

File: exec/csnex/Extensions.cs [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+            for (i = 0; i < self.Length && i < rhs.Length; i++) {
+                if (self[i] > rhs[i]) {
+                    return 1;
+                } else if (self[i] < rhs[i]) {
+                    return -1;
+                }
+            }
+            if (i == self.Length && i == rhs.Length) {
+                return 0; // The lhs and rhs are proven to be equal to one another.
+            } else if (i  == self.Length) {
```
Comment (ghewgill) [inline exec/csnex/Extensions.cs:31]: Fix spacing please.

**Review by ghewgill — CHANGES_REQUESTED** [review]

File: exec/csnex/Extensions.cs (hunk 3 above)
Comment (gitlarryf) [reply exec/csnex/Extensions.cs]: Opps, sorry.  Good catch.  Fixed it.

**Pull request merged**